### PR TITLE
Textarea v2

### PR DESCRIPTION
**Pull request by phgrey**
- Moon.Design.Form.Textarea
Screenshot 2023-01-18 at 09 48 23

```
  alias Moon.Design.Form.Textarea
  alias Moon.Design.Form.Addon.Btn
 ...
  <Textarea><Btn>Hit me!</Btn></Textarea>
```

### CODE REVIEW

File: lib/moon/design/form/addon/btn.ex [hunk 1]
```diff
@@ -0,0 +1,20 @@
+defmodule Moon.Design.Form.Addon.Btn do
```
Comment (margusp2heathmont) [inline lib/moon/design/form/addon/btn.ex:1]: 1. Why it is Addon.Btn? Would Form.SubmitActionButton or something similar make more sense?
2. Did this come from Design system and is agreed or it is some arbitary change?

Comment (phgrey) [reply lib/moon/design/form/addon/btn.ex:1]: 2. it's not from the design system, but I have seen 2 simillar problems - eye on a password and save inside textrare. they are both fixable in terms of headless ui by the same approach
1. It's not about a submit or any action at all. it's just a shorthand to put button "visually inside" textarea as i've seen on a screen.

Comment (phgrey) [reply lib/moon/design/form/addon/btn.ex:1]: 2. the same way is suitable for adding some counters "18/255 symbols" and so on

Comment (phgrey) [reply lib/moon/design/form/addon/btn.ex:1]: 2. had a short discussion with @dkireev - he agrredd to leave this "add-on system" as an experiment

File: lib/moon/design/form/addon/btn.ex [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  @moduledoc false
+
+  use Moon.StatelessComponent, slot: "default"
+
+  alias Moon.Design.Button
+
+  prop(attrs, :map, default: %{})
```
Comment (margusp2heathmont) [inline lib/moon/design/form/addon/btn.ex:8]: No

File: lib/moon/design/form/addon/btn.ex (hunk 1 above)
Comment (margusp2heathmont) [reply lib/moon/design/form/addon/btn.ex:1]: Ok, but please do not use Srt Fn Nms, Bcs Thy r hrd t rd.

File: lib/moon/design/form/addon/btn.ex (hunk 2 above)
Comment (phgrey) [reply lib/moon/design/form/addon/btn.ex:8]: the same way i do plan to make a PasswordInput - It is supposed to use TextInput component & some new Addon.Eye

**Pull request closed without merging**